newIDE/app/src/EventsSheet/EventsTree/Renderers/ForEachChildVariableEvent.js

```diff
@@ -0,0 +1,341 @@
+      editingIterableVariableName: false,
+      anchorEl: null,
+    });
+  };
+
+  render() {
+    const forEachChildVariableEvent = gd.asForEachChildVariableEvent(
+      this.props.event
+    );
+    const valueIteratorName = forEachChildVariableEvent.getValueIteratorVariableName();
+    const keyIteratorName = forEachChildVariableEvent.getKeyIteratorVariableName();
+    const iterableName = forEachChildVariableEvent.getIterableVariableName();
+
+    return (
+      <div
+        style={styles.container}
+        className={classNames({
+          [executableEventContainer]: true,
+        })}
+      >
+        <div style={{ marginLeft: '5px' }}>
+          <span
+            className={classNames({
+              [disabledText]: this.props.disabled,
+            })}
+            tabIndex={0}
+          >
+            <Trans>For every child in </Trans>
+            <span
+              className={classNames({
+                [selectableArea]: true,
+                [instructionParameter]: true,
+                [nameAndIconContainer]: true,
+                scenevar: true,
+              })}
+              style={styles.variableContainer}
+              onClick={e => this.edit('iterable', e.currentTarget)}
+              onKeyPress={event => {
+                if (shouldActivate(event)) {
+                  this.edit('iterable', event.currentTarget);
+                }
+              }}
+            >
+              <img
+                className={icon}
+                src="res/types/scenevar.png"
+                alt="Scene variable"
+              />
+              {iterableName.length !== 0 ? (
+                <span>{iterableName}</span>
+              ) : (
+                <span className="instruction-missing-parameter">
+                  <Trans>{`<Select a variable>`}</Trans>
+                </span>
+              )}
+            </span>
+            <Trans>, store the child in variable </Trans>
+            <span
+              className={classNames({
+                [selectableArea]: true,
+                [instructionParameter]: true,
+                [nameAndIconContainer]: true,
+                scenevar: true,
+              })}
+              style={styles.variableContainer}
+              onClick={e => this.edit('iteratorValue', e.currentTarget)}
+              onKeyPress={event => {
+                if (shouldActivate(event)) {
+                  this.edit('iteratorValue', event.currentTarget);
+                }
+              }}
+            >
+              <img
+                className={icon}
+                src="res/types/scenevar.png"
+                alt="Scene variable"
+              />
+              {valueIteratorName.length !== 0 ? (
+                <span>{valueIteratorName}</span>
+              ) : (
+                <span>
+                  <Trans>{`<Select a variable> (optional)`}</Trans>
+                </span>
+              )}
+            </span>
+            <Trans>, </Trans>
```

Comment:
You can put a single `<Trans>` from "For every child in" until the last comma. This will generated a translation string like "For every child in {0}, store the child in variable {1},", which can be translated with the full context (and with the proper order, which is not necessarily the same according to the language). You could actually remove the `br` and make the whole thing translated as a single sentence. The new line does not seem really necessary to me :)

Reply:
Oh I didn't knew the Trans was that smart.

> The new line does not seem really necessary to me :)

Well, it doesn't break the line automatically so on smaller screens you cannot use or see the last variable field.

Reply:
> Well, it doesn't break the line automatically so on smaller screens you cannot use or see the last variable field.

Add to the style (next to the marginLeft):
```
    whiteSpace: 'normal',
    wordWrap: 'break-word',
```

which restore the usual HTML behavior of divs :) (white-space was set to `nowrap` for some reasons, I can't remember now, in the events tree)